data/plugins/language_c.lua

```diff
@@ -1,6 +1,11 @@
 -- mod-version:3
 local syntax = require "core.syntax"
 
+-- integer suffix combinations as a regex
+local isuf = [[(?:[lL][uU]|ll[uU]|LL[uU]|[uU][lL]\b|[uU]ll|[uU]LL|[uU]|[lL]\b|ll|LL)?]]
```

Comment:
You can probably simplify this by enabling case insensitiveness with like `(?i:this IS case INSENSITIVE)`.

Reply:
for some odd reason, the two Ls in long long have to have the same case for the literal to be valid

Reply:
if it can't be fixed easily, let's just keep this as is.

Reply:
What about the `\b`? Why is it needed only in some cases?

Reply:
@thacuber2a03 can you answer this question? I will merge if this conversation is resolved.

Reply:
oh, yeah, sorry
about the `\b`, I have it there to avoid having an l alongside another one with a different case be highlighted
which, honestly, I think it's overkill, but compilers actually don't compile programs with stuff like `UlL` or `Ll`

Reply:
in whatever case, I'm not sure whether I care that much anymore, so you can merge with or without that big ass regex

Reply:
@takase1121 almost forgot the mention